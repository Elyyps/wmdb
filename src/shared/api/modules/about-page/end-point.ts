import { generateAboutPageDummyData } from "./generate-dummy-data";

export const generateAboutPageData = generateAboutPageDummyData;
