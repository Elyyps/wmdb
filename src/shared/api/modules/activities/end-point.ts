import { generateActivitiesDummyData } from "./dummy-data";

export const generateActivitiesData = generateActivitiesDummyData;
