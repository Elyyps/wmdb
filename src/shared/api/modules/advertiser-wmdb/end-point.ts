import { generateAdvertiserDummyData } from "./dummy-data";
export const generateAdvertiserData = generateAdvertiserDummyData;
