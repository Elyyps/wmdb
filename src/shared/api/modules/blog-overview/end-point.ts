import { generateBlogOverviewDummyData, generateCards } from "./dummy-data";
import { ICompanyOutingCard } from "@app/api/core/company-outing-card";

export const generateBlogOverviewData = generateBlogOverviewDummyData;
const amountOfCards = 50;
export const getCardsWithPagination = (skip: number, take: number): ICompanyOutingCard[] => {
  let cards: ICompanyOutingCard[] = generateCards(amountOfCards);
  cards = cards.slice(skip, take + skip);

  return cards;
};
