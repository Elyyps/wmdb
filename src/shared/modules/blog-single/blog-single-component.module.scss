.blog-single {
  display: flex;
  flex-direction: column;
  &-body {
    p {
      color: black;
      margin-top: 0;
    }
    h4 {
      color: black;
      font-size: 16px;
      margin-bottom: 0;
    }

    h1 {
      font-size: 34px;
    }
  }
  &-body-text {
    * {
      padding: 0;
    }
    border-bottom: 1px solid #6a6a6a;
  }
  &-list {
    margin: 0;
    padding: 0;
    list-style: none;
    display: flex;
    align-items: center;
    color: $gray;
    font-size: 14px;
    margin: 0 -18px;
    li {
      padding: 0 18px;
      position: relative;
      line-height: 1;
      display: flex;
      > * {
        color: $gray;
        margin-right: 10px;
        svg {
          fill: $gray;
        }
      }
      &:not(:first-child) {
        &:after {
          content: "";
          position: absolute;
          width: 1px;
          height: 15px;
          background-color: $gray;
          display: block;
          left: 0;
          top: 50%;
          transform: translateY(-50%);
          margin-right: 2px;
        }
      }
    }
  }
}
