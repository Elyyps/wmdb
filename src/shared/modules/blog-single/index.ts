export * from "./blog-single.component";
