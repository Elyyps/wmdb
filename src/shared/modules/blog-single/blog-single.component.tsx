import * as React from "react";
import styles from "./blog-single-component.module.scss";
import { ImageComponent } from "@app/core/image";
import { BodyTextComponent } from "@app/core/bodytext";
import { IBlogSingle } from "@app/api/modules/blog-single/blog-single";
import { BlogSingleFormComponent } from "@app/core/blog-single-form";
import { ModuleSectionComponent } from "@app/core/module-section";
import { Breadcrumb } from "@app/core/breadcrumb";

export interface IBlogSingleComponentProps {
  blogSingle: IBlogSingle;
}

const BlogSingleComponent = ({ blogSingle }: IBlogSingleComponentProps) => {
  const [numberOfReactions, setNumberOfReactions] = React.useState(0);

  return (
    <div className={styles["blog-single"]}>
      <div className="uk-container">
        <Breadcrumb breadcrumbs={blogSingle.path} />
        <ImageComponent src={blogSingle.headerImage} />
      </div>
      <ModuleSectionComponent paddingTop={"40px"} paddingBottom={"40px"} size="small">
        <div className={styles["blog-single-body"]}>
          <h1>{blogSingle.title}</h1>
          <p className={styles["blog-single-body-text"]}>
            <BodyTextComponent html={blogSingle.content} />
            <br />
            <br />
          </p>
          <br />

          <div className={styles["blog-single-form"]}>
            <h2>Reacties {numberOfReactions}</h2>
            <BlogSingleFormComponent onSubmit={() => setNumberOfReactions(numberOfReactions + 1)} />
          </div>
        </div>
      </ModuleSectionComponent>
    </div>
  );
};

export { BlogSingleComponent };
