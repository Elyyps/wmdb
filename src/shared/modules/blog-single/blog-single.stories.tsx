import * as React from "react";

import { storiesOf } from "@storybook/react";
import { withA11y } from "@storybook/addon-a11y";
import { withProvider } from "@app/util";
import { BlogSingleComponent } from "./blog-single.component";
import { generateBlogSingleData } from "@app/api/modules/blog-single/end-point";

storiesOf("BlogSingle", module)
  .addDecorator(withA11y)
  .addDecorator(withProvider)
  .add("Basic implementation", () => <BlogSingleComponent blogSingle={generateBlogSingleData()} />);
