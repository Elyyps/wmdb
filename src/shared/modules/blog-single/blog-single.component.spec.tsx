import React from "react";
import { shallow } from "enzyme";
import { BlogSingleComponent } from "./blog-single.component";
import { generateBlogSingleData } from "@app/api/modules/blog-single/end-point";

describe("[BlogSingle]", () => {
  it("should render component without crashing", () => {
    const renderedComponent = shallow(<BlogSingleComponent blogSingle={generateBlogSingleData()} />);
    expect(renderedComponent).toMatchSnapshot();
  });
});
