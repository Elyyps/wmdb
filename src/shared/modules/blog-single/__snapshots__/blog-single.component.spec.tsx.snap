// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[BlogSingle] should render component without crashing 1`] = `
<div
  className="blog-single"
>
  <div
    className="uk-container"
  >
    <Breadcrumb
      breadcrumbs={
        Array [
          Object {
            "title": "Home",
            "url": "/",
          },
          Object {
            "title": "Blogs",
            "url": "/blog-overview",
          },
          Object {
            "title": "Vakantiepret: spelletjes voor onderweg!",
            "url": "",
          },
        ]
      }
    />
    <ImageComponent
      src="img-blog.jpg"
    />
  </div>
  <ModuleSectionComponent
    paddingBottom="40px"
    paddingTop="40px"
    size="small"
  >
    <div
      className="blog-single-body"
    >
      <h1>
        Vakantiepret: spelletjes voor onderweg!
      </h1>
      <p
        className="blog-single-body-text"
      >
        <BodyTextComponent
          html="
  <p>Het is altijd leuk om op vakantie te gaan, maar je moet wel eerst nog maar op de vakantiebestemming aankomen. Misschien moet je uren in de auto zitten of een tijdje doorbrengen op een vliegveld. Zeker wanneer je met kinderen reist kan dat nog wel een hele opgave zijn, maar gelukkig kan je er ook een leuke tijd van maken. Vakantiepret begint namelijk al onderweg, want er zijn tal van leuke spelletjes die je dan kan gaan spelen. Onderstaande spellen zijn allemaal geschikt voor op reis en kunnen eenvoudig worden meegenomen in een rugzak, voor sommige spelletjes heb je zelfs helemaal niks nodig behalve een goed humeur.</p>
  <h5> <b>Tablet en smartphone</b> </h5>
  <p>Kinderen gaan mee met hun tijd en de meeste weten precies hoe een tablet of een smartphone werkt, soms beter dan de ouders. Je kan de tablet of smartphone dan ook meenemen op reis. Zorg ervoor dat er vooraf al leuke filmpjes op worden gezet of nieuwe spelletjes zijn gedownload. Vergeet trouwens ook niet om een koptelefoon mee te nemen, zodat niemand anders last heeft van de geluiden. Enkele leuke spelletjes zijn: Candy Crush en Angry Birds. </p>
  <h5> <b>Spelletjes voor in de auto</b> </h5>
  <p>Kinderen kunnen nog wel eens een zware tijd hebben in de auto, want ze moeten uren blijven zitten. Gelukkig zijn er allerlei leuke spelletjes die in de auto gespeeld kunnen worden.</p>
  <h5> <b>Kentekenplaten raden</b> </h5>

  <p>Het enige wat je nodig hebt voor een spelletje ‘kentekenplaten raden’ is een pen, papier en een goede concentratie. Jij en kinderen gaan namelijk tijdens de reis bekijken of je een buitenlands kenteken tegenkomt. En dan is aan het deelnemers om te raden uit welk land de auto komt of misschien zelfs wel welke stad. En de kentekens van diplomaten, die leveren extra veel punten op. Welke speler kan de meeste punten scoren en wint het spel? Trouwens, kom je een koninklijke stoet tegen dan is degene die dat het eerste zegt meteen de grote winnaar van het spel.</p>
  <h5> <b>Ik zie, ik zie wat jij niet ziet</b> </h5>
  <p>Dit is hoogstwaarschijnlijk wel het bekendste autospelletje ter wereld. Iedereen in de auto kan plezier beleven aan dit spel. Het is ook een ideale manier om de kinderen van de tablet of smartphone te krijgen en meer van de omgeving te laten genieten. Een van de spelers zegt ‘ik zie, ik zie wat jij niet ziet en de kleur is’. En dan mag de rest proberen te raden wat het is. Maar let op, je mag niet iets kiezen waar je allang langs bent gereden, want dat is natuurlijk niet zo eerlijk.</p>
  <h5> <b>Woordenslang</b> </h5>

  <p>Een woordenslang maken is niet alleen heel leuk, maar ook nog eens heel leerzaam. Het is een simpel spel. Allereerst zal er een onderwerp uitgekozen worden, bijvoorbeeld dieren of eten. Kies je voor eten dan mag de eerste deelnemer iets opnoemen, bijvoorbeeld ‘pizza’. De twee deelnemer moet dan een woord bedenken dat begint met de laatste letter van het vorige woord, bijvoorbeeld ‘aardappels’. En de eerstvolgende moet dan iets van eten verzinnen dat begint met de letter s. Hoelang kunnen jullie het volhouden? </p>
  <h5> <b>Wie ben ik?</b> </h5>
  <p>Een van de spelers in de auto kiest een bekend persoon uit of een kennis die iedereen kent. Alle spelers mogen een x-aantal vragen stellen, zodat zij op die manier de persoon kunnen raden. Zorg ervoor dat het niet te moeilijk zal worden, want het moet wel leuk blijven. Handige vragen zijn bijvoorbeeld: is het man of een vrouw, waar is hij of zij bekend van? </p>
  "
        />
        <br />
        <br />
      </p>
      <br />
      <div
        className="blog-single-form"
      >
        <h2>
          Reacties 
          0
        </h2>
        <WithFormik(InnerForm)
          onSubmit={[Function]}
        />
      </div>
    </div>
  </ModuleSectionComponent>
</div>
`;
